openedx/core/djangoapps/content_libraries/tests/test_content_libraries.py

```diff
@@ -872,6 +872,26 @@ def test_block_types(self, slug, library_type, constrained):
             assert len(types) > 1
 
 
+@elasticsearch_test
```

Comment:
When I run:

```
EDXAPP_RUN_BLOCKSTORE_TESTS=1 python -Wd -m pytest --ds=cms.envs.test openedx/core/lib/blockstore_api/ openedx/core/djangolib/tests/test_blockstore_cache.py openedx/core/djangoapps/content_libraries/tests/
```

I am getting the following errors:
```
========================================================================================= short test summary info ==========================================================================================
ERROR openedx/core/lib/blockstore_api/tests/test_blockstore_api.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.
ERROR openedx/core/djangolib/tests/test_blockstore_cache.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.
ERROR openedx/core/djangoapps/content_libraries/tests/test_content_libraries.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enab...
ERROR openedx/core/djangoapps/content_libraries/tests/test_libraries_index.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable...
ERROR openedx/core/djangoapps/content_libraries/tests/test_runtime.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.
ERROR openedx/core/djangoapps/content_libraries/tests/test_static_assets.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.
ERROR openedx/core/djangoapps/content_libraries/tests/test_views_lti.py - RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 7 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================================== 9 warnings, 7 errors in 5.91s =======================================================================================
```

Reply:
Ah oops.. looks like I can't have a waffle switch test in a `unittest.skipUnless` condition. Should be fixed with [f381375](https://github.com/openedx/edx-platform/pull/29779/commits/f381375f15f18f8c00af770fbb74ce2da0b2c277).

Also note: I erratically get an error from `ContentLibrariesXBlockTypeOverrideBlockstoreServiceTest.test_block_type_handler_1` which is unrelated to these changes, and has perplexed me greatly. I've tried addressing it, but haven't been able to locate it. It only happens when I run the full list of tests, and even then, it's flaky.

```bash
root@studio:/edx/app/edxapp/edx-platform# clear && EDXAPP_RUN_BLOCKSTORE_TESTS=1 python -Wd -m pytest --ds=cms.envs.test -vvv openedx/core/lib/blockstore_api/ openedx/core/djangolib/tests/test_blockstore_cache.py openedx/core/djangoapps/content_libraries/tests/
<snip>
=============================================================== FAILURES ================================================================
___________________________ ContentLibrariesXBlockTypeOverrideBlockstoreServiceTest.test_block_type_handler_1 ___________________________

self = <openedx.core.djangoapps.content_libraries.tests.test_content_libraries.ContentLibrariesXBlockTypeOverrideBlockstoreServiceTest testMethod=test_block_type_handler_1>
slug = 'block-wo-override', api_args = {}, expected_type = 'video'

    @ddt.data(*BLOCK_DATA)
    @ddt.unpack
    @patch("openedx.core.djangoapps.xblock.rest_api.views.LX_BLOCK_TYPES_OVERRIDE", {'video': 'alt-block'})
    @XBlock.register_temp_plugin(AltBlock, 'alt-block')
    def test_block_type_handler(self, slug, api_args, expected_type):
        # Check that the handler_url contains the block type override params
        block_key = self.create_block(f"handler-{slug}")
        response = self.client.get(
            URL_BLOCK_GET_HANDLER_URL.format(block_key=block_key, handler_name='student_view_user_state'),
            api_args,
        )
        handler_url = response.data['handler_url']
        parsed_url = urlparse(handler_url)
        parsed_qs = dict(parse_qsl(parsed_url.query))
        assert parsed_qs == api_args

        # Ensure the invoked handler hits the expected Block
        if expected_type == 'video':
            expected_response = {
                'all_sources': [],
                'duration': None,
                'encoded_videos': {
                    'youtube': {
                        'file_size': 0,
                        'url': 'https://www.youtube.com/watch?v=rE42zZ-3wNo',
                    }
                },
                'only_on_web': False,
                'saved_video_position': 0.0,
                'speed': None,
                'transcripts': {},
            }
        else:
            expected_response = {
                "id": f'lb:CL-TEST:handler-{slug}:video:handler-{slug}',
                "block_type": 'video',
                "override_type": "<class 'xblock.internal.AltBlockWithMixins'>",
            }
        response = self.client.post(handler_url).json()
        # Can't match the Transcripts download URL exactly, but we can check that it's there and roughly correct
        if 'transcripts' in response:
            assert f"lb:CL-TEST:handler-{slug}:video:handler-{slug}" in response['transcripts']['en']
            del response['transcripts']['en']
>       assert response == expected_response
E       AssertionError: assert {'all_sources': [],\n 'duration': None,\n 'encoded_videos': {'youtube': {'file_size': 0,\n
                'url': 'https://www.youtube.com/watch?v=3_yD_cEKoCk'}},\n 'only_on_web': False,\n 'saved_video_position': 0.0,\n 'speed': None,\n 'transcripts': {}} == {'all_sources': [],\n 'duration': None,\n 'encoded_videos': {'youtube': {'file_size': 0,\n
                'url': 'https://www.youtube.com/watch?v=rE42zZ-3wNo'}},\n 'only_on_web': False,\n 'saved_video_position': 0.0,\n 'speed': None,\n 'transcripts': {}}
E         Common items:
E         {'all_sources': [],
E          'duration': None,
E          'only_on_web': False,
E          'saved_video_position': 0.0,
E          'speed': None,
E          'transcripts': {}}
E         Differing items:
E         {'encoded_videos': {'youtube': {'file_size': 0, 'url': 'https://www.youtube.com/watch?v=3_yD_cEKoCk'}}} != {'encoded_videos': {'youtube': {'file_size': 0, 'url': 'https://www.youtube.com/watch?v=rE42zZ-3wNo'}}}
E         Full diff:
E           {
E            'all_sources': [],
E            'duration': None,
E            'encoded_videos': {'youtube': {'file_size': 0,
E         -                                 'url': 'https://www.youtube.com/watch?v=rE42zZ-3wNo'}},
E         ?                                                                         ^ ^^^^^^^^
E         +                                 'url': 'https://www.youtube.com/watch?v=3_yD_cEKoCk'}},
E         ?                                                                         ^^^^^^ ^ ++
E            'only_on_web': False,
E            'saved_video_position': 0.0,
E            'speed': None,
E            'transcripts': {},
E           }

openedx/core/djangoapps/content_libraries/tests/test_content_libraries.py:1094: AssertionError
----------------------------------------------------------- Captured log call -----------------------------------------------------------
ERROR    openedx.core.djangoapps.safe_sessions.middleware:middleware.py:179 SafeCookieData BWC parse error: 'wbv8u3zafylz2gpbnmryci6ewdfs48qc'.
WARNING  openedx.core.djangoapps.xblock.runtime.runtime:runtime.py:108 thirdparty handlers are not supported by this runtime for XBlock <class 'xblock.internal.VideoBlockWithMixins'>.
<snip>
======================================================== short test summary info ========================================================
FAILED openedx/core/djangoapps/content_libraries/tests/test_content_libraries.py::ContentLibrariesXBlockTypeOverrideBlockstoreServiceTest::test_block_type_handler_1
```

Reply:
Hmm.. the CI is failing to start the mongo service :( I'm going to have to wait for a new commit or rebase to re-trigger the tests at another time

Reply:
If I remember correctly, the tests are run in random order so that tests do not develop dependencies on others. Could there be something being set up in a different test that is causing this failure when that test is run before?

Reply:
Probably? I'm way over budget on this task though, so haven't had time to nut it out.
The issue only happens for me when running the blockstore service integration tests (which aren't run on CI), and so I'm not too concerned about this flaky test.

Reply:
I see. Okay, to leave it as it is then.